source/text/vss-strings.ads

```diff
@@ -98,6 +98,241 @@ package VSS.Strings is
       Right : Virtual_String) return Boolean;
    --  Compare two strings in binary order of code points.
 
+   --  function "&"
+   --    (Left  : Virtual_String;
+   --     Right : Virtual_String) return Virtual_String;
+   --  function "&"
+   --    (Left  : Virtual_String;
+   --     Right : VSS.Characters.Virtual_Character) return Virtual_String;
+   --  function "&"
+   --    (Left  : VSS.Characters.Virtual_Character;
+   --     Right : Virtual_String) return Virtual_String;
+   --  function "&"
+   --    (Left  : VSS.Characters.Virtual_Character;
+   --     Right : VSS.Characters.Virtual_Character) return Virtual_String;
+   --  --  Concatenation operator for virtual string and virtual characters.
+
+   procedure Clear (Self : in out Virtual_String'Class);
+   --  Remove all data.
+
+   procedure Append
+     (Self : in out Virtual_String'Class;
+      Item : VSS.Characters.Virtual_Character);
+   --  Append given abstract character to the end of the string.
+
+   --  function Append
+   --    (Self : Virtual_String'Class;
+   --     Item : VSS.Characters.Virtual_Character) return Virtual_String;
+   --  --  Append given abstract character to the end of the string and returns
+   --  --  result.
+   --
+   --  procedure Append
+   --    (Self : in out Virtual_String'Class;
+   --     Item : Virtual_String'Class);
+   --  --  Append given string to the end of the string.
+   --
+   --  function Append
+   --    (Self : Virtual_String'Class;
+   --     Item : Virtual_String'Class) return Virtual_String;
+   --  --  Append given string to the end of the string and returns result.
+
+   --  procedure Prepend
+   --    (Self : in out Virtual_String'Class;
+   --     Item : VSS.Characters.Virtual_Character);
+   --  --  Prepend given character to the begin of the string.
+   --
+   --  procedure Prepend
+   --    (Self : in out Virtual_String'Class;
+   --     Item : Virtual_String'Class);
+   --  --  Prepend given string to the begin of the string.
+   --
+   --  procedure Insert
+   --    (Self     : in out Virtual_String'Class;
+   --     Position : VSS.Strings.Cursors.Abstract_Character_Cursor'Class;
+   --     Item     : VSS.Characters.Virtual_Character);
+   --  procedure Insert
+   --    (Self     : in out Virtual_String'Class;
+   --     Position : VSS.Strings.Cursors.Abstract_Character_Cursor'Class;
+   --     Item     : Virtual_String'Class);
+   --  procedure Insert
+   --    (Self     : in out Virtual_String'Class;
+   --     Position : VSS.Strings.Cursors.Abstract_Segment_Cursor'Class;
+   --     Item     : VSS.Characters.Virtual_Character);
```

Comment:
Do we really need `Insert` for segment cursors? Where do inserted character go? After the segment end or after the first character of segment beginning?

Could be not clear for the user.

Reply:
Insert operations insert data at position pointing by the cursor and moves existing data after inserted one. It is not important is it single character element or segment element.